pkg/controllers/machine/disruption/drift.go

```diff
@@ -78,3 +78,26 @@ func (d *Drift) Reconcile(ctx context.Context, _ *v1alpha5.Provisioner, machine
 	// Requeue after 5 minutes for the cache TTL
 	return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
 }
+
+// isDrifted will check if a machine is drifted from the fields in the provisioner.Spec and
+// the cloudprovider
+func (d *Drift) isDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (bool, error) {
+	cloudProviderDrifted, err := d.cloudProvider.IsMachineDrifted(ctx, machine)
+	if err != nil {
+		return false, err
+	}
+	staticFieldsDrifted := areStaticFieldsDrifted(provisioner, machine)
+
+	return cloudProviderDrifted || staticFieldsDrifted, nil
```

Comment:
nit: could in-line if you want
```suggestion
	return cloudProviderDrifted || areStaticFieldsDrifted(provisioner, machine), nil
```